src/wallet/rpcdump.cpp

```diff
@@ -151,9 +151,12 @@ UniValue importprivkey(const JSONRPCRequest& request)
         CKeyID vchAddress = pubkey.GetID();
         {
             pwallet->MarkDirty();
-            // We don't know which corresponding address will be used; label them all
+
+            // We don't know which corresponding address will be used; label all new addresses and preserve the rest
             for (const auto& dest : GetAllDestinationsForKey(pubkey)) {
-                pwallet->SetAddressBook(dest, strLabel, "receive");
+                bool has_label = pwallet->mapAddressBook.find(dest) != pwallet->mapAddressBook.end() && !pwallet->mapAddressBook[dest].name.empty();
```

Comment:
I was thinking in something like
```cpp
if (request.params[1].isNull()) {
    const auto& it = pwallet->mapAddressBook.find(dest);
    if (it != pwallet->mapAddressBook.end()) strLabel = it->second;
}
pwallet->SetAddressBook(dest, strLabel, "receive");
```

Reply:
Hm, I see. Yeah.. this behavior seems better. Thanks!
Updating this and the test.